### PR TITLE
Feature/kb patch request

### PR DESCRIPTION
**Pull request by kirstybutler**

### CODE REVIEW

**Review by raynelle03 — APPROVED** [review]

File: shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/MealsSDK.kt [hunk 1]
```diff
@@ -16,6 +16,10 @@ class MealsSDK {
         return api.postMeal(meal)
     }
 
+    @Throws(Exception::class) suspend fun patchMeal(id: String, meal: Meal): Meal {
+        return api.patchMeal(meal, id)
+    }
```
Comment (Gmiller290488) [inline shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/MealsSDK.kt]: we can just pass through the meal here.
And then get out the id from the Meal object

File: shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt [hunk 2]
```diff
@@ -23,8 +27,12 @@ class MealApi {
...
     suspend fun postMeal(meal: Meal): Meal {
         return httpClient.post(MEALS_ENDPOINT) {
-            url(MEALS_ENDPOINT)
-            contentType(ContentType.Application.Json)
+            body = meal
+        }
+    }
+
+    suspend fun patchMeal(meal: Meal, id: String): Meal {
+        return httpClient.patch("$MEALS_ENDPOINT/$id") {
```
Comment (Gmiller290488) [inline shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt]: and then use `$meal.id` here

File: shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/MealsSDK.kt (hunk 1 above)
Comment (kirstybutler) [reply shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/MealsSDK.kt]: Hmm, so how should ID be passed through the httpClient fun?

Comment (kirstybutler) [reply shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/MealsSDK.kt]: Nvm just saw your other comment ^^

**Review by dsmith42 — APPROVED** [review]
LGTM

File: shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt (hunk 2 above)
Comment (kirstybutler) [reply shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt]: Yaaas thanks!

Comment (kirstybutler) [reply shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt]: Muuuuch better

**Review by Gmiller290488 — APPROVED** [review]
Noice

**Review by roger-tan — APPROVED** [review]

**Pull request merged**